scripts/autorelease.js

```diff
@@ -0,0 +1,218 @@
+                   https://github.com/settings/tokens
+                   Check all repo
+
+`;
+
+const { appendFileSync, chmodSync, mkdtempSync, writeFileSync, mkdirSync } = require('fs');
+const { join, resolve, dirname } = require('path');
+const { spawnSync } = require('child_process');
+const { tmpdir } = require('os');
+const { inc } = require('semver');
+const https = require('https');
+
+// Import environment variables
+const {
+  AUTORELEASE_BASE = 'master',
+  AUTORELEASE_KEY,
+  AUTORELEASE_TOKEN,
+  HOME,
+  WERCKER_GIT_BRANCH,
+  WERCKER_GIT_DOMAIN,
+  WERCKER_GIT_OWNER,
+  WERCKER_GIT_REPOSITORY,
+} = process.env;
+
+// Process arguments
+const {
+  SCRIPT
+} = (() => {
+  const options = {};
+  const argv = process.argv;
+
+  for (let x = 2; x < argv.length; ++x) {
+    const [arg, ...values] = argv.split('=');
+    const value = values.join('=');
+
+    switch (arg) {
+      case '--help':
+        console.log(`usage: ${USAGE}`);
+        process.exit(1);
+        break;
+      case '--script':
+        options.script = values.length ? value : argv[++x];
+        break;
+      default:
+        console.error(`Argument "${arg}" not recognized`);
+        process.exit(1);
+    }
+  }
+
+  return options;
+})();
+
+// Determine which release we're going to use
+const [FEATURE, RELEASE_TYPE = 'patch'] = WERCKER_GIT_BRANCH.split('/');
+
+if (FEATURE !== 'release') {
+  console.error('This is not a release branch, so not creating a release');
+  process.exit(0);
+}
+
+const packageJson = require(resolve('package.json'));
+const NEXT_VERSION = inc(packageJson.version, RELEASE_TYPE);
+
+if (NEXT_VERSION)
+  console.log(`Preparing ${RELEASE_TYPE} release from ${packageJson.version} to ${NEXT_VERSION}`);
+else {
+  console.log(
+    `Cannot create ${RELEASE_TYPE} release from ${packageJson.version} to ${NEXT_VERSION}`
+  );
+  process.exit(1);
+}
+
+const AUTORELEASE_DIR = mkdtempSync(join(tmpdir(), 'autorelease-'));
+const ID_RSA = resolve(AUTORELEASE_DIR, 'id_rsa');
+const SSH_CONFIG = `${HOME}/.ssh/config`;
+const KNOWN_HOSTS = `${HOME}/.ssh/known_hosts`;
+
+/**
+ * Run spawnSync
+ *
+ * @param {string} command The command to run
+ * @param {string[]} args Arguments for the command
+ * @param {object} options The options
+ *
+ * @return {object} The result of spawnSync
+ */
+function exec(command, args, options = {}) {
+  console.log(command, args);
+  const result = spawnSync(command, args, options);
+  console.log(result.stdout.toString('ascii'));
+  console.log(result.stderr.toString('ascii'));
+  if (result.error) throw result.error;
+  if (result.status) process.exit(1);
+  return result;
+}
+
+mkdirSync(dirname(SSH_CONFIG), [0o600]);
+
+// Create the SSH deploy key
+writeFileSync(ID_RSA, AUTORELEASE_KEY.replace(/\\n/g, '\n'));
+chmodSync(ID_RSA, 0o400);
+appendFileSync(SSH_CONFIG, `
+Host autorelease
+  User git
+  HostName ${WERCKER_GIT_DOMAIN}
+  IdentityFile ${ID_RSA}`, {
+    mode: 0o600
+  });
```

Comment:
This line should be dedented with the parent so that the start and end are the same indentation. Below is how I probably would have written it.

```js
appendFileSync(SSH_CONFIG, `
Host autorelease
  User git
  HostName ${WERCKER_GIT_DOMAIN}
  IdentityFile ${ID_RSA}`,
  { mode: 0o600 }
);
```